tests/sentry/runner/commands/test_backup.py

```diff
@@ -0,0 +1,56 @@
+import atexit
+import os
+
+import pytest
+from click.testing import CliRunner
+from django.db import IntegrityError
+
+from sentry.runner.commands.backup import export, import_
+from sentry.testutils import CliTestCase
+from sentry.utils import json
+
+tmp_backup_filename = "test_backup.json"
+
+
+class BackupTest(CliTestCase):
+    command = export
+
+    def test_export(self):
+        rv = self.invoke(tmp_backup_filename)
+        assert rv.exit_code == 0, rv.output
+        assert os.path.exists(tmp_backup_filename)
+
+
+# Avoid using CliTestCase/TestCase since it wraps everything in an atomic function,
+# which creates an error with hybrid cloud fixture protect_user_deletion
+@pytest.mark.django_db
+def test_import():
+    rv = CliRunner().invoke(import_, tmp_backup_filename)
+    assert rv.exit_code == 0, rv.output
+
+
+@pytest.mark.django_db
+def test_import_duplicate_key():
+    # Adding an element with the same key as the last item in the backed up file
+    # to force a duplicate key violation exception
+    with open(tmp_backup_filename) as backup_file:
+        contents = json.load(backup_file)
+        duplicate_key_item = contents[-1]
+        duplicate_key_item["pk"] += 1
+        contents.append(duplicate_key_item)
+    with open(tmp_backup_filename, "w") as backup_file:
+        backup_file.write(json.dumps(contents))
+    rv = CliRunner().invoke(import_, tmp_backup_filename)
+    assert (
+        rv.output
+        == ">> Are you restoring from a backup of the same version of Sentry?\n>> Are you restoring onto a clean database?\n>> If you're doing both already, file a bug report, it's probably our fault\n"
+    )
+    assert isinstance(rv.exception, IntegrityError)
+    assert rv.exit_code == 1
+
+
+# cleanup backup file once test suite finishes
+@atexit.register
```

Comment:
Is this a normal way we do cleanup in other tests? I would expect this to be in some `def teardown` hook on a `TestCase`.

Reply:
It is not, the problem here is I wanted to bypass using `TestCase` because of the fixture `protect_user_deletion`. It's causing issues when trying to set that up and giving me errors `You can't execute queries until the end of the 'atomic' block`.

Reply:
Okay, let's put that in a comment to explain why we're doing something weird here.

Reply:
put that earlier in a comment on why we're not using `TestCase` or `CliTestCase` but elaborated a bit so it's more clear

Reply:
Ah, gotcha. Yes, perfect.

Reply:
this is not the normal way that cleanup should be done -- this should be using `NamedTemporaryFile` in a context manager

this is causing this test to be flaky currently: https://github.com/getsentry/sentry/actions/runs/4969861749/jobs/8893355143

Reply:
Thanks, ticketed at https://github.com/getsentry/sentry/issues/49351.